src/main/java/org/cactoos/collection/CollectionEnvelope.java

```diff
@@ -147,4 +147,48 @@ public String toString() {
         return this.col.value().toString();
     }
 
+    @Override
+    public final boolean equals(final Object other) {
+        return new UncheckedScalar<>(
+            new And(
+                new Or(
+                    () -> new InheritanceLevel(
+                        other.getClass(), Collection.class
+                    ).value() > -1,
+                    () -> new InheritanceLevel(
+                        other.getClass(), CollectionEnvelope.class
```

Comment:
@Vatavuk I believe this isn't necessary because `CollectionEnvelope` *is a* `Collection` already

Reply:
@llorllale the end result is that we will compare only classes that are related to `CollectionEnvelope`

Reply:
@Vatavuk We should be comparing `Collection`s here, not `CollectionEnvelope`s.

I think I understand the problem now: `InheritanceLevel` does not check if a class implements an interface (maybe it shouldn't).

We're missing tests that compare `CollectionEnvelope` against regular JDK `Collection`s. I believe they will fail with the current implementation. So we either need to a) add an `instanceof` (against our principles) or b) build a new type that checks if a class implements/extends an interface, or c) modify `InheritanceLevel` to take into account (b).

Please leave a puzzle for this and we'll merge.